### PR TITLE
Add list section to web hierarchy section

### PR DESCRIPTION
**Pull request by nataliafitzgerald**
Adds bulleted lists, numbered lists, nested lists, and link lists to the DM.

## Additions
- Bulleted lists
- Numbered lists
- Nested lists (bulleted)
- Nested lists (numbered)
- Link lists

## Removals
-

## Changes
-

## Testing

-

## Review

- @marteki @Scotchester @jimmynotjim 

[Preview this PR without the whitespace changes](?w=0)

## Screenshots
screencapture-0-0-0-0-4000-brand-guidelines-typography-html-1519423142199

## Notes

-

## Todos

-

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: brand-guidelines/typography.md [hunk 1]
```diff
@@ -259,6 +258,155 @@ If you have received financial aid offers, use this tool to compare school data
...
+
+<div class="content-33 content-first">
+
+#### Nested list (bulleted)
+
+* Avenir Next Regular
+* 16px / 8px
+* Margin top: 10px
+* Margin bottom: 15px
+* Nested bullets are flush left with the preceeding line of text
```
Comment (marteki) [inline brand-guidelines/typography.md]: Typo: Extra 'e' in the word "preceding"

File: brand-guidelines/typography.md [hunk 2]
```diff
@@ -259,6 +258,155 @@ If you have received financial aid offers, use this tool to compare school data
...
+
+<div class="content-33 content-first">
+
+#### Nested list (numbered)
+
+* Avenir Next Regular
+* 16px / 8px
+* Margin top: 10px
+* Margin bottom: 15px
+* Nested bullets are flush left with the preceeding line of text
```
Comment (marteki) [inline brand-guidelines/typography.md]: Typo: Extra 'e' in the word "preceding"

File: brand-guidelines/typography.md [hunk 3]
```diff
@@ -259,6 +258,155 @@ If you have received financial aid offers, use this tool to compare school data
...
+<div class="content-33 content-first">
+
+#### Nested list (bulleted)
+
+* Avenir Next Regular
+* 16px / 8px
+* Margin top: 10px
+* Margin bottom: 15px
+* Nested bullets are flush left with the preceeding line of text  
+
```
Comment (marteki) [inline brand-guidelines/typography.md]: Nice to have, but not necessary to add for approval: a description of the nested bullets for this type being round circles

File: brand-guidelines/typography.md [hunk 4]
```diff
@@ -259,6 +258,155 @@ If you have received financial aid offers, use this tool to compare school data
...
+<div class="content-33 content-first">
+
+#### Nested list (numbered)
+
+* Avenir Next Regular
+* 16px / 8px
+* Margin top: 10px
+* Margin bottom: 15px
+* Nested bullets are flush left with the preceeding line of text 
+
```
Comment (marteki) [inline brand-guidelines/typography.md]: Nice to have, but not necessary to add for approval: a description of the nested bullets for this type being solid squares

**Review by marteki — CHANGES_REQUESTED** [review]
Two typos to be fixed. Two lines that could be added if you agree they might be valuable (and if not, that's cool).

File: brand-guidelines/typography.md (hunk 1 above)
Comment (nataliafitzgerald) [reply brand-guidelines/typography.md]: Fixed.

File: brand-guidelines/typography.md (hunk 2 above)
Comment (nataliafitzgerald) [reply brand-guidelines/typography.md]: Fixed.

**Review by marteki — APPROVED** [review]

**Comment by Scotchester** [comment]
A few questions/comments:

1. Similar to what I suggested to Jimmy in the Capital Framework docs, is it necessary to break out the nested styles into separate sections, or could we just include a nested list within each of the main list sections?
1. Is it worth showing a nested numbered list? (I.e., an example where both parent and child lists are numbered.)
1. The "16px / 8px" is somewhat confusing to me. Typically that kind of construction would represent type size / leading, but the 8px here is intended to represent the extra margin between list items, correct? I think it'd be good to be more explicit about that.

**Comment by nataliafitzgerald** [comment]
@Scotchester 
I've captured your recommendations and will submit a separate PR next week to address them. I think your points 1 and 3 make sense. In terms of including the nested numbered list, I can certainly include it but I haven't seen a use case for this in our content. Have you? I just want to make sure that the patterns we're including are something we'd want people to use.

Do you feel comfortable with merging what we have here as a first step?

**Pull request merged**